### PR TITLE
Refs #28909 -- Simplifed code using unpacking generalizations.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Comment by pope1ni** [comment]
LGTM.

File: django/forms/widgets.py [hunk 1]
```diff
@@ -958,9 +958,8 @@ def get_context(self, name, value, attrs):
         year_choices = [(i, str(i)) for i in self.years]
         if not self.is_required:
             year_choices.insert(0, self.year_none_value)
-        year_attrs = context['widget']['attrs'].copy()
         year_name = self.year_field % name
-        year_attrs['id'] = 'id_%s' % year_name
+        year_attrs = {**context['widget']['attrs'], 'id': 'id_%s' % year_name}
```
Comment (timgraham) [inline django/forms/widgets.py]: The `foo_attrs` variables could be removed.

**Comment by pope1ni** [comment]
@PaulSD - Can you provide a simple example that shows this?

This should work the same as before:

- `tpl.setdefault()` only set the value if it wasn't already in `tpl`.
- Unpacking `tpl` at the end of the redefinition of `tpl` is equivalent to calling `tpl.copy()` (which is a shallow copy).
- Unpacking generalisations allow for duplicate keys in dict literals with the last value overriding any earlier values.

So in this case the value of `NAME` in the original `tpl` dict will still override the default value in the redefinition.

**Comment by PaulSD** [comment]
Sorry, you are right.  I had a different issue and I mis-interpreted what I was seeing here.  This does work a expected.

**Pull request merged**